Services/Style/Content/Container/class.ContainerDBRepository.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ */
+
+namespace ILIAS\Style\Content\Container;
+
+/**
+ * This repo stores infos on repository objects that are using booking managers as a service
+ * (resource management).
+ *
+ * @author Alexander Killing <killing@leifos.de>
+ */
+class ContainerDBRepository
+{
+    const TABLE_NAME = 'sty_rep_container';
+
+    /**
+     * @var \ilDBInterface
+     */
+    protected $db;
+
+    public function __construct(\ilDBInterface $db)
+    {
+        $this->db = $db;
+    }
+
+    public function updateReuse(int $ref_id, bool $reuse)
+    {
+        $db = $this->db;
+
+        $db->replace(
+            self::TABLE_NAME,
+            [
+                "ref_id" => ["integer", $ref_id]
+            ],
+            [
+                "reuse" => ["integer", (int) $reuse]
+            ]
+        );
+    }
+
+    public function readReuse(int $ref_id) : bool
+    {
+        $db = $this->db;
+
+        $set = $db->queryF(
+            "SELECT * FROM " . self::TABLE_NAME .
```

Comment:
```suggestion
            "SELECT reuse FROM " . self::TABLE_NAME .
```

Or even better:

```sql
SELECT EXISTS(SELECT 1 FROM ... WHERE ...) cnt`
````

And:

```php
return $rec && (int) $rec['cnt'] === 1;
``